synapse/push/pusherpool.py

```diff
@@ -107,6 +107,7 @@ async def add_pusher(
         data: JsonDict,
         profile_tag: str = "",
         enabled: bool = True,
+        device_id: Optional[str] = None,
```

Comment:
Does it make sense for this to be optional? If this version of the source code is adding a new pusher, shouldn't it know which device it's adding it for?

Reply:
If we're eg adding it with the admin API, there won't be a device ID.